src/transport_interface/transport_interface_test.c

```diff
@@ -421,26 +522,137 @@ static void prvSendRecvCompareFunc( void * pParam )
     for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
     {
         /* Initialize the test data buffer. */
-        prvInitializeTestData( pTrasnportTestBufferStart, testBufferSize );
+        prvInitializeTestData( pTransportTestBufferStart, testBufferSize );
 
         /* Send the test data to the server. */
         if( pThreadParameter->xResult == true )
         {
-            pThreadParameter->xResult = prvTransportSendData( pTestTransport, pNetworkContext, pTrasnportTestBufferStart,
+            pThreadParameter->xResult = prvTransportSendData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
+                                                              testBufferSize );
+        }
+
+        /* Receive the test data from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvTransportRecvData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
                                                               testBufferSize );
         }
 
+        /* Compare the test data received from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvVerifyTestData( pTransportTestBufferStart,
+                                                           testBufferSize,
+                                                           TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH );
+        }
+
+        if( ( pThreadParameter->stopFlag == true ) || ( pThreadParameter->xResult == false ) )
+        {
+            break;
+        }
+
+        #if ( TRANSPORT_TEST_PRINT_DEBUG_PROGRESS == 1 )
+            /* Output information to indicate the test is running. */
+            UNITY_OUTPUT_CHAR( '.' );
+        #endif
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+#ifdef TRANSPORT_TEST_EXECUTE_WRITEV_TESTS
+/**
+ * @brief Thread function of the writev/receive/compare test.
+ */
+static void prvWritevRecvCompareFunc( void * pParam )
+{
+    threadParameter_t * pThreadParameter = pParam;
+    uint8_t * pTransportTestBufferStart =
+        &( pThreadParameter->transportTestBuffer[ TRANSPORT_TEST_BUFFER_PREFIX_GUARD_LENGTH ] );
+    uint32_t testBufferSize = 0;
+    NetworkContext_t * pNetworkContext = pThreadParameter->pNetworkContext;
+    TransportOutVector_t pTransportTestVectorArray[ 5U ];
+    size_t ioVecCount;
+
+    pThreadParameter->xResult = true;
+
+    for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
+    {
+        if( testBufferSize % 5 == 0 )

```

Comment:
Would it make sense to replace this with assigning a variable to be equal to the modulo result and then use a for loop? That way it's something like

```
if (testBufferSize % 5 == 0)
    splitCount = 5;
 /* else ifs */
 /* We split the data based off of the testBufferSize */
 for( uint32_t i = 0; i < testBufferSize; i++){
            pTransportTestVectorArray[ i ].iov_base = &pTransportTestBufferStart[ testBufferSize / ( 5 * i) ];
            pTransportTestVectorArray[ i ].iov_len = testBufferSize / 5;
 }
 ```
 
 I know that there can be good reasons for doing it this way, hence why I wanted to ask.